kernel/src/scan/mod.rs

```diff
@@ -274,16 +274,16 @@ mod tests {
     use std::path::PathBuf;
 
     use super::*;
+    use crate::client::sync::SyncInterface;
```

Comment:
Similar to other question -- rust packaging is smart enough to not make sync client a prod dependency merely because tests need it? Or do we need some additional annotations here?

(several more below)

Reply:
update: this is a test-only config so we're fine.